### DISCUSSION TITLE
Sonoff basic r2 hardware version 1.3

### DISCUSSION THREAD

**Discussion by Khaleidsg** [arendst/Tasmota#20627] [Q&A] [answered]
Hello 

I installed Tasmota on Sonoff basic r2, 
Hardware version 1.3
But to know the GPIO pins ?

Thank you 
![Screenshot_20240127_152522_Facebook](https://github.com/arendst/Tasmota/assets/100121678/b19d1657-aca7-45e9-ae4e-8de35ccdbc5f)

**Comment by barbudor** [comment]
As stated in the [docs](https://tasmota.github.io/docs/Templates/), you should look on the [template repository](https://templates.blakadder.com/)

**Reply by Khaleidsg** [reply]
I want to know the hardware gpio locations on the pcb!?

**Reply by barbudor** [reply, marked as answer]
Some are already clearly written like GPIO2, GPIO1 (TX), GPIO3 (RX)
For the others, here is the pinout of the ESP8266 : https://tasmota.github.io/docs/Pinouts/#esp82xx-based
GPIO0, 12 and 13 that are connected to components (button, ledn relay) should be easy to follow and identify

For the others they may not be any accessible places

Sonoff Basic are not the best device for DIY stuff

**Comment by Noschvie** [comment]
> Sonoff Basic are not the best device for DIY stuff

Is the "SONOFF MINI R2" a recommendable one? Thanks!

**Reply by sfromis** [reply]
Not for DIY stuff where you want expand with external connections. They are not designed for projects, only as a Wifi controlled relay.

**Reply by barbudor** [reply]
There are more appropriate boards for tinkering
Starting with Nodemcu /dev kit to play with sensors
Or boards with already PSU, relays, broken out gpios

**Comment by lalo-uy** [comment]
Get a Wemos/Lolin D1 mini
